packages/@aws-cdk/aws-dax/lib/cluster.ts

```diff
@@ -0,0 +1,348 @@
+   */
+  readonly clusterName: string;
+}
+
+/**
+ * A DAX cluster
+ *
+ * @resource AWS::DAX::Cluster
+ */
+export class Cluster extends cdk.Resource implements ICluster {
+  /**
+   * Import an existing cluster to the stack from its attributes.
+   */
+  public static fromClusterAttributes(scope: cdk.Construct, id: string, attrs: ClusterAttributes): ICluster {
+    return new ImportedCluster(scope, id, attrs);
+  }
+
+  /**
+   * The VPC associated with the cluster.
+   */
+  public readonly vpc: IVpc;
+
+  /**
+   * The security group which protects access to the DAX cluster
+   */
+  public readonly securityGroup: SecurityGroup;
+
+  /**
+   * Manage the allowed network connections for the cluster with Security Groups.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * The role of the cluster
+   */
+  public readonly role: Role;
+
+  /**
+   * The underlying Cfn resource for the cluster
+   * @attribute
+   */
+  public readonly cluster: CfnCluster;
+
+  /**
+   * The endpoint at which the cluster can be accessed
+   * @attribute
+   */
+  public readonly clusterDiscoveryEndpoint: string;
+
+  /**
+   * The ARN of the cluster
+   * @attribute
+   */
+  public readonly clusterArn: string;
+
+  /**
+   * The name of the cluster
+   * @attribute
+   */
+  public readonly clusterName: string;
+
+  /**
+   * IAM policies associated with the cluster
+   */
+  protected policies: Policy[];
+
+  protected id: string;
+
+  constructor(scope: cdk.Construct, id: string, props: ClusterProps) {
+    super(scope, id);
+    this.id = id;
+    this.vpc = props.vpc;
+    this.clusterName = props.clusterName;
+
+    // Define a group for telling Elasticache which subnets to put cache nodes in.
+    const subnetGroup = new CfnSubnetGroup(this, `${id}-subnet-group`, {
+      description: `List of subnets used for dax cache ${id}`,
+      subnetIds: this.vpc.privateSubnets.map(subnet => subnet.subnetId),
+    });
+
+    // The security group that defines network level access to the cluster
+    this.securityGroup = new SecurityGroup(this, `${id}-security-group`, { vpc: this.vpc });
+
+    this.connections = new Connections({
+      securityGroups: [this.securityGroup],
+      defaultPort: new Port({
+        protocol: Protocol.TCP,
+        fromPort: 8111,
+        toPort: 8111,
+        stringRepresentation: 'TCP',
+      }),
+    });
+
+    this.role = new Role(this, `${id}-role`, {
+      assumedBy: new ServicePrincipal('dax.amazonaws.com'),
+      description: 'Role that allows DynamoDB Accelerator to talk to DynamoDB',
+    });
+
+    // The parameters for this cluster
+    let parameterGroup = props.parameterGroup;
+
+    if (!parameterGroup) {
+      parameterGroup = new ParameterGroup(this, '${id}-parameters', {
+        parameters: {
+          'query-ttl-millis': cdk.Duration.minutes(5).toMilliseconds().toString(),
+          'record-ttl-millis': cdk.Duration.minutes(5).toMilliseconds().toString(),
+        },
+      });
+    }
```

Comment:
```suggestion
    const parameterGroup = props.parameterGroup
      ?? new ParameterGroup(this, '${id}-parameters', {
        parameters: {
          'query-ttl-millis': cdk.Duration.minutes(5).toMilliseconds().toString(),
          'record-ttl-millis': cdk.Duration.minutes(5).toMilliseconds().toString(),
        },
      });
```

The `ParameterGroup` class looks fairly under-modeled..